src/main/java/com/gmail/chernii/oleksii/dao/CompanyDAO.java

```diff
@@ -0,0 +1,63 @@
+package com.gmail.chernii.oleksii.dao;
+
+import com.gmail.chernii.oleksii.entity.Company;
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j;
+
+import javax.persistence.EntityManager;
+
+/**
+ * Created by Space on 16.04.2019.
+ */
+@Log4j
+@AllArgsConstructor
+public class CompanyDAO implements DAO<Company> {
+    private EntityManager entityManager;
+
+    @Override
+    public void insert(Company company) {
+        try {
+            entityManager.getTransaction().begin();
+            entityManager.persist(company);
+            entityManager.getTransaction().commit();
+        } catch (RuntimeException e) {
+            if (entityManager != null) {
```

Comment:
Why not put the same part of the code into a separate method?

Reply:
what do you mean?

Reply:
sorry, i've found